.github/workflows/tenzir.yaml

```diff
@@ -83,10 +83,60 @@ jobs:
       run-python: ${{ steps.configure.outputs.run-python }}
       run-python-package: ${{ steps.configure.outputs.run-python-package }}
     steps:
+      - name: Configure ssh-agent
+        uses: webfactory/ssh-agent@v0.8.0
+        with:
+          ssh-private-key: ${{ secrets.TENZIR_PLUGINS_DEPLOY_KEY }}
       - name: Checkout
         uses: actions/checkout@v4
         with:
+          submodules: true
           fetch-depth: 0
+          persist-credentials: false
+      - name: Generate a token
+        if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+        id: generate_token
+        uses: actions/create-github-app-token@v1
+        with:
+          app-id: ${{ vars.TENZIR_TESTAPP_APP_ID }}
+          private-key: ${{ secrets.TENZIR_TESTAPP_PRIVATE_KEY }}
+          owner: ${{ github.repository_owner }}
+      - name: Test submodule update
```

Comment:
The name seems to be very innocuous compared what it actually does. Maybe make it a bit clearer?